app/src/main/java/org/sagebionetworks/research/mpower/studyburst/StudyBurstActivity.kt

```diff
@@ -268,16 +289,52 @@ class StudyBurstActivity : AppCompatActivity(), StudyBurstAdapterListener {
      * StudyBurstAdapterListener function, called when a task icon in the RecyclerView is selected.
      */
     override fun itemSelected(item: StudyBurstTaskInfo) {
+
         val uuid = studyBurstViewModel.createScheduleTaskRunUuid(item.schedule?.guid)
-        taskLauncher.launchTask(this, item.task.identifier, uuid)
-                .observe(this, Observer {
-                    when(it?.state) {
-                        LAUNCH_ERROR -> {
-                            Toast.makeText(this,
-                                    "Error launching  " + item.task.identifier,
-                                    Toast.LENGTH_LONG).show()
+
+        if (item.task.identifier == WALK_AND_BALANCE) {
```

Comment:
Walk and balance bug fix, had to do a special case to match how the TrackingMenuFragment was launching it correctly.  The root cause is that TaskLauncher was built without a way for Activities to launch RS tasks, only fragments.